saga/seata-saga-engine-store/src/main/java/io/seata/saga/engine/config/DbStateMachineConfig.java

```diff
@@ -39,14 +45,33 @@
  */
 public class DbStateMachineConfig extends DefaultStateMachineConfig implements DisposableBean {
 
+    private static final Logger LOGGER = LoggerFactory.getLogger(DbStateMachineConfig.class);
+
     private DataSource dataSource;
     private String applicationId;
     private String txServiceGroup;
     private String tablePrefix = "seata_";
     private String dbType;
     private SagaTransactionalTemplate sagaTransactionalTemplate;
-    private boolean rmReportSuccessEnable = ConfigurationFactory.getInstance().getBoolean(ConfigurationKeys.CLIENT_REPORT_SUCCESS_ENABLE, DEFAULT_CLIENT_REPORT_SUCCESS_ENABLE);
-    private boolean sagaBranchRegisterEnable = ConfigurationFactory.getInstance().getBoolean(ConfigurationKeys.CLIENT_SAGA_BRANCH_REGISTER_ENABLE, DEFAULT_CLIENT_SAGA_BRANCH_REGISTER_ENABLE);
+    private boolean rmReportSuccessEnable = DEFAULT_CLIENT_REPORT_SUCCESS_ENABLE;
+    private boolean sagaBranchRegisterEnable = DEFAULT_CLIENT_SAGA_BRANCH_REGISTER_ENABLE;
+    private String sagaJsonParser = DEFAULT_SAGA_JSON_PARSER;
+
+
+    public DbStateMachineConfig() {
+        try {
+            Configuration configuration = ConfigurationFactory.getInstance();
+            if (configuration != null) {
+                this.rmReportSuccessEnable = configuration.getBoolean(ConfigurationKeys.CLIENT_REPORT_SUCCESS_ENABLE, DEFAULT_CLIENT_REPORT_SUCCESS_ENABLE);
+                this.sagaBranchRegisterEnable = ConfigurationFactory.getInstance().getBoolean(ConfigurationKeys.CLIENT_SAGA_BRANCH_REGISTER_ENABLE, DEFAULT_CLIENT_SAGA_BRANCH_REGISTER_ENABLE);
```

Comment:
`ConfigurationFactory.getInstance()` to `configuration`

Reply:
done